allensdk/brain_observatory/behavior/data_files/demix_file.py

```diff
@@ -0,0 +1,66 @@
+import json
+from typing import Dict, Union
+from pathlib import Path
+
+import h5py
+from cachetools import cached, LRUCache
+from cachetools.keys import hashkey
+
+import pandas as pd
+
+from allensdk.internal.api import PostgresQueryMixin
+from allensdk.brain_observatory.behavior.data_files import DataFile
+
+
+def from_json_cache_key(cls, dict_repr: dict):
+    return hashkey(json.dumps(dict_repr))
+
+
+def from_lims_cache_key(cls, db, ophys_experiment_id: int):
```

Comment:
Should little general methods like this also go in `cachetools`? Would this ever be used anywhere else?

Reply:
I will leave this up to @njmei since he implemented it

Reply:
I kept these separate since in theory the parameters to `from_lims_cache_key` may change depending on the `DataObject`/`DataFile`.